### ISSUE TITLE
https://wpt.fyi/test-runs shows commits out of order

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt.fyi#615]
https://wpt.fyi/test-runs current shows this:
![image](https://user-images.githubusercontent.com/498917/46397977-0f8cb680-c6f4-11e8-8def-af7f4cc6e9c7.png)

I found it surprising that there are stable runs for two consecutive commits and suspected a bug, but actually it's just a problem with the order. Probably because the Edge and Safari results for 67152fdecd arrived many hours later than the others, that row has a later date. The other rows with Buildbot results (f15449622a and 63641fb5fe) also appear out of order compared to the commit time order.

https://wpt.fyi/test-runs?label=taskcluster does match the commit order currently, but that's probably not guaranteed because the Taskcluster runs for two consecutive commits will be racing to finish.

**Comment by foolip** [comment]
Even though it wouldn't fix the order in all cases, just sorting by time of earliest arrived results rather than latest arrived results would fix most of this.

**Comment by jugglinmike** [comment]
[We discussed this previously and landed on "sort by `test_start` time" as a rough approximation](https://github.com/web-platform-tests/wpt.fyi/issues/310). I think the issue here isn't about when the results are uploaded, but when the builds were started. On Sauce Labs, we don't start collecting from Safari until we've finished with Edge. Those results have a delayed `test_start`, and that's probably to blame here.

**Comment by Hexcles** [comment]
Thanks for the information, @jugglinmike . We are already doing what @foolip suggested in the first comment, but I wasn't aware of the SauceLabs situation. I guess the reason why we only start Safari after Edge is capacity/quota?

**Comment by jugglinmike** [comment]
Capacity means we can't run all the builds in parallel, but it doesn't mean we can't schedule collection for the browsers to start and finish at roughly the same time. That would be possible if we interleaved the segments for Safari and Edge. Even then, though, the results would be arriving long after the Firefox and Chrome results (from both Buildbot and TaskCluster).

**Comment by Hexcles** [comment]
*Arriving* much later isn't an issue, as long as they *start* as soon as possible, the list recent runs will be ordered more correctly (closer to commit history).

I don't know how much change it needs and how risky it is to start Sauce and Edge at the same time. If it's practical, it'd be nice to have.

**Comment by foolip** [comment]
I don't think we should be making any changes to results collection to fix this problem, as it still wouldn't be entirely fixed. Rather, I think we should use the commit date. That's a bit of work of course, but this isn't a very serious problem, so just using the earliest start time for any run for a given commit and sorting by that would be a decent workaround until we have some other use for git commit data.

**Comment by Hexcles** [comment]
Oh sorry I didn't read @foolip 's first comment carefully enough. We are sorting by test start time, not arrival time.

**Comment by foolip** [comment]
Note that in https://github.com/web-platform-tests/wpt.fyi/issues/667#issuecomment-430784319 the reordering of commits actually presented a bit of a problem, in that it seemed like the problem was fixed and then regressed, but it was just the commits changing order.
